### DISCUSSION TITLE
Docker for synology

### DISCUSSION THREAD

**Discussion by melbasuonyy** [Taxel/PlexTraktSync#800] [Support]
Hello, 

Can I use this with synology, it's not available in docker search Also it takes a very long time to sync my tv shows collections (10,000+) series (days) is this normal

Thanks

**Comment by andrelung** [comment]
Not sure if it's the right way, but this was my path:

1. ssh into my synology diskstation
2. sudo docker pull ghcr.io/taxel/plextraktsync:latest
3. open the webinterface, open docker 
4. click on the "Image"-tab
5. select "/taxel/plextraktsync" and click on start
6. click on container, details and move to tab "Terminal"
7. it took some time - but at some point the configuration showed up
8. (do the configuration)

then it basically worked. Furthermore I wanted to create a recurring task. To do so:
1. open "Settings" on the diskstation
2. click on tasks (all the way down)
3. create a planned task with a custom script
4. give it some name and assign the user root (this is needed to command docker)
5. create a schedule, for example daily at a certain time
6. click on task settings 
7. add the command "docker start plextraktsync -a" (without "")
8. save and done

Lastly you will get the notification, that the container stopped unexpectedly. To remove that execute the following one time:
1. start the container from the webinterface
2. stop it immediatly
3. (this will change the containers configuration for the field "enabled"; method is based on: https://github.com/Taxel/PlexTraktSync/discussions/917#discussioncomment-2782020)


Hope it helps!

Regarding the time: on a ds920+ my first full sync took 16 min 18.9 seconds (via log-entry). Every next sync is about ~10 minutes.

**Reply by glensc** [reply]
> sudo docker pull ghcr.io/taxel/plextraktsync:main

why are you using `main` branch? where has this information has been falsely spread again?
you should use `latest` or a more specific version tag, it can also be `0.21` for 0.21.x versions.

using `main` as un-released code is not supported

**Reply by brandontravis** [reply]
FWIW, this is the only way I could get this working in DSM 7 using docker, except for I changed main to latest

Also:
"it took some time - but at some point the configuration showed up"

All I had to do was click into the terminal and hit enter. Config showed up immediately after. I haven't followed @andrelung's instructions for creating a sync task yet, as my initial sync is still happening but I have a fairly large library so I'll attempt afterwards.

**Reply by glensc** [reply]
will you explain why or where you took the `main`, so could <del>compain</del> inform there too?

**Reply by brandontravis** [reply]
Not sure why the hostility, but I use latest for pretty much all docker images unless instructed otherwise. Your response specifically said to use latest so I did. I'm not complaining about anything. I was simply saying that the above steps helped me get finally get this up and running. I encountered issues getting this setup on my Synology attempting any other instructions that I had found.

**Reply by glensc** [reply]
you used `main` initially, that's why I'm asking why, because our docs say to use `latest` not `main`. as people come up and still use `main`, I want to know why, or from where is that coming from.

so, why don't you answer or explain?

ps: I'm not here about the problem (can't comment on it), I only asked about tag name in this discussion thread.

**Reply by glensc** [reply]
> so, why don't you answer or explain? (about there `main` came from)

**Reply by andrelung** [reply]
Hey @glensc ! Sorry for the late reply - I've been totally sick. Using main wasn't intentionally, sorry for that. I think I've copied it from somewhere. Looking back at my browser history I think it came from here:
https://github.com/Taxel/PlexTraktSync/pkgs/container/plextraktsync
after clicking on the accepted answer here:
https://github.com/Taxel/PlexTraktSync/discussions/96

**Reply by andrelung** [reply]
I also changed my instructions. Thanks for the hint!

**Reply by glensc** [reply]
>  I think it came from here:
> - https://github.com/Taxel/PlexTraktSync/pkgs/container/plextraktsync

that's something we can't change. sigh. so just common sense for users.

**Reply by glensc** [reply]
edited the accepted answer from https://github.com/Taxel/PlexTraktSync/discussions/96

**Reply by Tiwas** [reply]
Thanks! Awesome guide :)

Anyone have any experience with how long it takes before it's visible on trakt? I just ran mine, and nothing seems to have happened...

Will it only sync for files still on my plex server, or will it also sync deleted episodes/movies?

**Reply by scoiattolo49** [reply]
> Anyone have any experience with how long it takes before it's visible on trakt? I just ran mine, and nothing seems to have happened...

Immediately after the sync has completed. 
During scrobbling you should be able to see in Trakt that the movie/show is being played in real time.

> Will it only sync for files still on my plex server, or will it also sync deleted episodes/movies?

Only the ones that are still present in your Plex server are part of a current sync. 
Any previous Trakt history (if any) will not be deleted from Trakt.

**Reply by Tiwas** [reply]
> > Will it only sync for files still on my plex server, or will it also sync deleted episodes/movies?
> 
> Only the ones that are still present in your Plex server are part of a current sync. Any previous Trakt history (if any) will not be deleted from Trakt.

Ok, thanks. I was hoping it would somehow sync deleted episodes/movies as well, as I don't have unlimited storage. I usually delete stuff once I've watched it.

**Reply by glensc** [reply]
- https://github.com/Taxel/PlexTraktSync/issues/374
  - https://github.com/Taxel/PlexTraktSync/pull/1363
  - https://github.com/Taxel/PlexTraktSync/pull/1377
  
  it's implemented, but not documented, both have docs needed łabel, so be the one who documents!

**Reply by Tiwas** [reply]
> * [Feature request: Remove deleted items from Trakt collection #374](https://github.com/Taxel/PlexTraktSync/issues/374)
>   
>   * [Feature: Add sync.plex_to_trakt.clear_collected config (for Movies only) #1363](https://github.com/Taxel/PlexTraktSync/pull/1363)
>   * [Feature: Add sync.plex_to_trakt.clear_collected config (for Episodes) #1377](https://github.com/Taxel/PlexTraktSync/pull/1377)
>   
>   it's implemented, but not documented, both have docs needed łabel, so be the one who documents!

I think there might have been a misunderstanding. I don't need anything deleted from trakt.

Here's my scenario:
* I watch an episode/move
* I delete said file from plex
* I run PlexTraktSync at midnight

The preferred outcome: The episode/movie is marked as watched in trakt.

**Reply by scoiattolo49** [reply]
> Here's my scenario:
> 
> * I watch an episode/move
> * I delete said file from plex
> * I run PlexTraktSync at midnight
> 
> The preferred outcome: The episode/movie is marked as watched in trakt.

If you delete the files before the sync takes place, those files will not be synced.

You need to enable scrobbling to mark in real time a file as played in Trakt:
https://github.com/Taxel/PlexTraktSync#watch

**Reply by Tiwas** [reply]
> > Here's my scenario:
> > 
> > * I watch an episode/move
> > * I delete said file from plex
> > * I run PlexTraktSync at midnight
> > 
> > The preferred outcome: The episode/movie is marked as watched in trakt.
> 
> If you delete the files before the sync takes place, those files will not be synced.
> 
> You need to enable scrobbling to mark in real time a file as played in Trakt: https://github.com/Taxel/PlexTraktSync#watch

Thanks! Does this mean the server has to run always? I've set it up per the instructions here, but I guess I can edit the config and leave it running on startup :)

**Reply by Tiwas** [reply]
> 1. start the container from the webinterface
> 2. stop it immediatly
> 3. (this will change the containers configuration for the field "enabled"; method is based on: [Synology reports unexpected behaviour #917 (comment)](https://github.com/Taxel/PlexTraktSync/discussions/917#discussioncomment-2782020))

Do you know how to revert back to it not stopping after running the sync? Or how I can edit the config? I would like to try the scrobbling function, but I have no idea how to edit the config when the container just stops.

**Reply by scoiattolo49** [reply]
The method described in this discussion to run Plextraktsync is a bit unusual and IMHO it's not the way a container should be run. I suggest you use a different and more standard method so the container can be set up to run both Watch and sync modes.

I use the image you find here:
https://github.com/linuxserver-labs/docker-plextraktsync
that is already configured to run both in Watch and sync modes (you use the crontab file to set how often the sync takes place)
If you read the Readme you find easy installation instructions for both CLI and Docker Compose methods

**Reply by Tiwas** [reply]
> The method described in this discussion to run Plextraktsync is a bit unusual and IMHO it's not the way a container should be run. I suggest you use a different and more standard method so the container can be set up to run both Watch and sync modes.
> 
> I use the image you find here: https://github.com/linuxserver-labs/docker-plextraktsync that is already configured to run both in Watch and sync modes (you use the crontab file to set how often the sync takes place) If you read the Readme you find easy installation instructions for both CLI and Docker Compose methods

Thanks. I installed this and made a crontab entry (learned about @reboot) to start the watch function. Works like a charm 🤩
